tools/topology/topology2/cavs-nocodec.conf

```diff
@@ -76,10 +76,13 @@ Define {
 	SSP0_MIXER_SOURCE_2		'mixin.16.1'
 	SSP0_MIXER_SINK_2		'mixout.16.1'
 	SSP0_MIXER_SOURCE_3		'mixin.21.1'
+	SSP1_ENABLED			"false"
 }
 
 # override defaults with platform-specific config
 IncludeByKey.PLATFORM {
+	"tgl"	"platform/intel/tgl.conf"
+	"adl"	"platform/intel/tgl.conf"
```

Comment:
I am having a bit of a heartburn with this PR.

Is the pinmux is a problem on one instance of the MTL RVP, or the MTL SOC in general? It's got to be the former, there is not general incompatibility between SSP1 and DMIC, it's just that the configuration of ONE board that has a restriction.

So rather than having a blanket removal for SSP1, I only only remove SSP1 on specific targets, and use udev rules or a tplg_name kernel option to work-around this limitation.

Reply:
@plbossart Let's extinguish the fire that burns your heart, how about generating two topology targets, see updated code.
- sof-mtl-nocodec-ssp0-ssp2.tplg, use this one for MTL RVP, only SSP0 and SSP2 are enabled
- sof-mtl-nocodec.tplg, use this for other MTL device on which all SSPs are enabled.

I think this should address your concern that there are boards have/haven't the pin-mux issue. And configurations are all in the cmake, and only needed if we want to disable SSP1.

Reply:
@plbossart good for you now ?